ui/components/app/snaps/snap-ui-renderer/snap-ui-renderer.test.js

```diff
@@ -112,582 +118,760 @@ function renderInterface(
 describe('SnapUIRenderer', () => {
   beforeEach(() => {
     jest.resetAllMocks();
+    const mockUseScrollValue = {
+      isScrollable: true,
+      scrollToBottom: jest.fn(),
+      ref: { current: {} },
+      onScroll: jest.fn(),
+    };
+    mockUseScrollRequired.mockReturnValue(mockUseScrollValue);
   });
 
-  it('renders loading state', () => {
-    const { container } = renderInterface(null);
-
-    expect(container.getElementsByClassName('pulse-loader')).toHaveLength(1);
-    expect(container).toMatchSnapshot();
-  });
-
-  it('renders basic UI', () => {
-    const { container, getByText, getRenderCount } = renderInterface(
-      Box({ children: Text({ children: 'Hello world!' }) }),
-    );
-
-    expect(getByText('Hello world!')).toBeDefined();
-    expect(getRenderCount()).toBe(1);
-    expect(container).toMatchSnapshot();
-  });
-
-  it('renders footers', () => {
-    const { container, getByText } = renderInterface(
-      Container({
-        children: [
-          Box({ children: Text({ children: 'Hello world!' }) }),
-          Footer({ children: Button({ children: 'Foo' }) }),
-        ],
-      }),
-      { useFooter: true },
-    );
-
-    expect(getByText('Foo')).toBeDefined();
-    expect(container).toMatchSnapshot();
-  });
+  describe('Basic Rendering', () => {
```

Comment:
The change in nesting makes this diff very difficult to read. Maybe move it to a separate PR, and only add the required changes here?

Reply:
Yeah, I can do that